Doppler.BigQueryMicroservice/Controllers/BigQueryController.cs

```diff
@@ -41,14 +58,38 @@ public async Task<ActionResult> GetAllowedEmails(string accountName)
         [HttpPut("/big-query/{accountName}/allowed-emails")]
         public async Task<ActionResult> SaveAllowedEmails(string accountName, AllowedEmails model)
         {
-            var user = await _userRepository.GetUserByEmail(accountName);
-            if (user == null)
+            try
             {
-                return NotFound();
+                bool notified = false;
+                var user = await _userRepository.GetUserByEmail(accountName);
+                if (user == null)
+                {
+                    return NotFound();
+                }
+
+                var results = await _userAccessByUserRepository.MergeEmailsAsync(user.IdUser, model.Emails);
+                var newEmails = results.Where(a => a.Action == "INSERT").Select(e => e.Email).ToList();
+                var template = _emailSettings.Value.BigQueryInvitationTemplateId[model.Languaje.ToLower()];
+                if (newEmails.Count > 0)
+                {
+                    try
+                    {
+                        notified = await _emailSender.SendWithTemplateAsync(template, newEmails, model.Emails);
+                    }
+                    catch (System.Exception ex)
+                    {
+                        _logger.LogError(ex, "Error in method SendWithTemplateAsync, service {service}", nameof(EmailSender));
+                        notified = false;
+                    }
+                }
+
+                return Ok(new { saved = true, notified = notified });
```

Comment:
I think that `notified` logic is not clear, let us see the scenarios:

1. No inserted emails (no error) → `{ saved: true, notified: false }` 
2. Inserted emails with no error in sending → `{ saved: true, notified: true }` 
3. Inserted emails with error in sending → `{ saved: true, notified: false }` 

So, you get `notified: false` in two different scenarios, with error in sending and without error in sending.

Of course, you can go further and include information about the inserted, removed, or kept emails, but probably it is not necessary. But probably it is not necessary, what do you think about removing all not strictly necessary logic from here?